.github/workflows/pr-build.yml

```diff
@@ -20,18 +41,113 @@ jobs:
+          environment-file: air-quality-backend/conda/utility_environment.yml
           auto-activate-base: false
-          activate-environment: etl
+          activate-environment: utility-env
          
       - name: Run lint check
         run: python -m flake8 --append-config air-quality-backend/.flake8 air-quality-backend
 
-      - name: Run unit tests
-        run: python -m pytest --cov=src --cov=scripts --cov-report=xml:air-quality-backend/tests/coverage/test_report.xml air-quality-backend/tests
+  etl:
+    needs: detect-change-scope
+    if: ${{ needs.detect-change-scope.outputs.etl == 'true' }}
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -el {0}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup miniconda
+        uses: conda-incubator/setup-miniconda@v3
+        with:
+          channels: conda-forge
+          environment-file: air-quality-backend/conda/etl_environment.yml
+          auto-activate-base: false
+          activate-environment: etl-dev
+
+      - name: Run etl unit tests
+        run: python -m pytest --cov=src --cov=scripts air-quality-backend/tests/etl_tests
         
+      - name: Rename coverage file
+        run: mv .coverage .coverage.etl
+
+      - name: Upload coverage
+        uses: actions/upload-artifact@master
+        with:
+          name: etl_test_report
+          path: .coverage.etl
+
+  api:
+    needs: detect-change-scope
+    if: ${{ needs.detect-change-scope.outputs.api == 'true' }}
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -el {0}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup miniconda
+        uses: conda-incubator/setup-miniconda@v3
+        with:
+          channels: conda-forge
+          environment-file: air-quality-backend/conda/api_environment.yml
+          auto-activate-base: false
+          activate-environment: api-dev
+         
+      - name: Run api unit tests
+        run: python -m pytest --cov=tests --cov=scripts air-quality-backend/tests/api_tests
+
+      - name: Rename coverage file
+        run: mv .coverage .coverage.api
+        
+      - name: Upload coverage
+        uses: actions/upload-artifact@master
+        with:
+          name: api_test_report
+          path: .coverage.api
+      
+            
+  combine-coverage:
+    needs: [detect-change-scope, etl, api]
+    if: ${{ needs.detect-change-scope.outputs.etl == 'true' && needs.detect-change-scope.outputs.api == 'true' }}
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -el {0}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup miniconda
+        uses: conda-incubator/setup-miniconda@v3
+        with:
+          channels: conda-forge
+          environment-file: air-quality-backend/conda/api_environment.yml
```

Comment:
This can probably be the utility env you created as we're just combining coverage

Reply:
This is actually why I created the utility-env, I just forgot to change it.  My bad XD